src/handler/promote.ts

```diff
@@ -0,0 +1,99 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda/trigger/api-gateway-proxy';
+import 'dotenv/config';
+import { cloneDeep } from 'lodash';
+import { TechRecordGet } from '../models/post';
+import { PromoteRecordRequestBody } from '../models/promote';
+import { SearchCriteria } from '../models/search';
+import { archiveOldCreateCurrentRecord, getBySystemNumberAndCreatedTimestamp, searchByCriteria } from '../services/database';
+import { getUserDetails } from '../services/user';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { validatePromoteErrors } from '../validators/promote';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Promote end point called');
+
+  const promoteErrors = validatePromoteErrors(event);
+
+  if (promoteErrors) {
+    return addHttpHeaders(promoteErrors);
+  }
+
+  const systemNumber = decodeURIComponent(event?.pathParameters?.systemNumber ?? '');
+  const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp ?? '');
+  const userDetails = getUserDetails(event.headers.Authorization ?? '');
+  const { reasonForPromoting } = JSON.parse(event.body as string) as PromoteRecordRequestBody;
+
+  logger.info(`Get from database with systemNumber ${systemNumber} and timestamp ${createdTimestamp}`);
+
+  const provisionalRecord: TechRecordGet = await getBySystemNumberAndCreatedTimestamp(systemNumber, createdTimestamp) as TechRecordGet;
+
+  logger.debug(`result is: ${JSON.stringify(provisionalRecord)}`);
+
+  if (!provisionalRecord || !Object.keys(provisionalRecord).length) {
+    return addHttpHeaders({
+      statusCode: 404,
+      body: `No record found matching systemNumber ${systemNumber} and timestamp ${createdTimestamp}`,
+    });
+  }
+
+  if (provisionalRecord.techRecord_statusCode !== 'provisional') {
```

Comment:
Probably best to have an enum for statusCodes

Reply:
Where should this go? Does the types package include enums or just declare locally? What do you think?

Reply:
I was doing them locally